### ISSUE TITLE
Bug: The security token included in the request is invalid

### ISSUE THREAD

**Issue by dimasusername** [aws/aws-sam-cli#3971]
### Description:
I did find a [similar issue](https://github.com/aws/aws-sam-cli/issues/2143) but was closed and despite new comments added is not re-opened.

Invoking an image runtime lambda locally produces the error in the title when accessing Secrets Manager.

### Steps to reproduce:
Create sample dockerized Lambda, try accessing Secrets Manager.

Invoke: `sam local invoke "Function" -e events/test_sm.json --profile *** --debug`

### Observed result:
```
2022-06-14 22:42:44,359 | Telemetry endpoint configured to be https://aws-serverless-tools-telemetry.us-west-2.amazonaws.com/metrics
2022-06-14 22:42:44,359 | Using config file: samconfig.toml, config environment: default
2022-06-14 22:42:44,359 | Expand command line arguments to:
2022-06-14 22:42:44,359 | --template_file=/home/***/code/***/***-***/.aws-sam/build/template.yaml --event=events/test_sm.json --function_logical_id=Function --no_event --layer_cache_basedir=/home/***/.aws-sam/layers-pkg --container_host=localhost --container_host_interface=127.0.0.1 
2022-06-14 22:42:44,359 | local invoke command is called
2022-06-14 22:42:44,363 | Collected default values for parameters: {}
2022-06-14 22:42:44,371 | Sam customer defined id is more priority than other IDs. Customer defined id for resource Function is Function
2022-06-14 22:42:44,371 | 0 stacks found in the template
2022-06-14 22:42:44,371 | Collected default values for parameters: {}
2022-06-14 22:42:44,378 | Sam customer defined id is more priority than other IDs. Customer defined id for resource Function is Function
2022-06-14 22:42:44,378 | 1 resources found in the stack 
2022-06-14 22:42:44,379 | Found Serverless function with name='Function' and ImageUri='function:latest'
2022-06-14 22:42:44,379 | --base-dir is not presented, adjusting uri /home/***/code/***/***-*** relative to /home/***/code/***/***-***/.aws-sam/build/template.yaml
2022-06-14 22:42:44,379 | --base-dir is not presented, adjusting uri . relative to /home/***/code/***/***-***/.aws-sam/build/template.yaml
2022-06-14 22:42:44,382 | Found one Lambda function with name 'Function'
2022-06-14 22:42:44,382 | Invoking Container created from function:latest
2022-06-14 22:42:44,382 | Environment variables overrides data is standard format
2022-06-14 22:42:44,382 | Loading AWS credentials from session with profile '***'
2022-06-14 22:42:44,389 | Code None is not a zip/jar file
Building image.................
2022-06-14 22:42:44,463 | Skip pulling image and use local one: function:rapid-1.50.0-x86_64.

2022-06-14 22:42:44,647 | Starting a timer for 300 seconds for function 'Function'
START RequestId: 27b5c17f-7b42-4dcf-85d2-3df84810bea1 Version: $LATEST
Skipped bootstraping TelemetryLog
Executing 'lambda.lambda_handler' in function directory '/function'
Error raised from handler method
{
  "errorMessage": "The security token included in the request is invalid.",
  "errorType": "Function<Aws::SecretsManager::Errors::UnrecognizedClientException>",
  "stackTrace": [
    "/usr/local/bundle/gems/aws-sdk-core-3.131.1/lib/seahorse/client/plugins/raise_response_errors.rb:17:in `call'",
    "/usr/local/bundle/gems/aws-sdk-core-3.131.1/lib/aws-sdk-core/plugins/checksum_algorithm.rb:111:in `call'",
    "/usr/local/bundle/gems/aws-sdk-core-3.131.1/lib/aws-sdk-core/plugins/jsonvalue_converter.rb:22:in `call'",
    "/usr/local/bundle/gems/aws-sdk-core-3.131.1/lib/aws-sdk-core/plugins/idempotency_token.rb:19:in `call'",
    "/usr/local/bundle/gems/aws-sdk-core-3.131.1/lib/aws-sdk-core/plugins/param_converter.rb:26:in `call'",
    "/usr/local/bundle/gems/aws-sdk-core-3.131.1/lib/seahorse/client/plugins/request_callback.rb:71:in `call'",
    "/usr/local/bundle/gems/aws-sdk-core-3.131.1/lib/aws-sdk-core/plugins/response_paging.rb:12:in `call'",
    "/usr/local/bundle/gems/aws-sdk-core-3.131.1/lib/seahorse/client/plugins/response_target.rb:24:in `call'",
    "/usr/local/bundle/gems/aws-sdk-core-3.131.1/lib/seahorse/client/request.rb:72:in `send_request'",
    "/usr/local/bundle/gems/aws-sdk-secretsmanager-1.62.0/lib/aws-sdk-secretsmanager/client.rb:1323:in `get_secret_value'",
    "/function/secrets_manager.rb:39:in `read_secret_json'",
    "/function/secrets_manager.rb:15:in `initialize'",
    "/function/lambda.rb:15:in `new'",
    "/function/lambda.rb:15:in `initialize'",
    "/function/lambda.rb:46:in `new'",
    "/function/lambda.rb:46:in `lambda_handler'",
    "/usr/local/bundle/gems/aws_lambda_ric-2.0.0/lib/aws_lambda_ric/lambda_handler.rb:28:in `call_handler'",
    "/usr/local/bundle/gems/aws_lambda_ric-2.0.0/lib/aws_lambda_ric.rb:81:in `run_user_code'",
    "/usr/local/bundle/gems/aws_lambda_ric-2.0.0/lib/aws_lambda_ric.rb:59:in `start_runtime_loop'",
    "/usr/local/bundle/gems/aws_lambda_ric-2.0.0/lib/aws_lambda_ric.rb:42:in `run'",
    "/usr/local/bundle/gems/aws_lambda_ric-2.0.0/lib/aws_lambda_ric/bootstrap.rb:35:in `bootstrap_handler'",
    "/usr/local/bundle/gems/aws_lambda_ric-2.0.0/lib/aws_lambda_ric/bootstrap.rb:8:in `start'",
    "/usr/local/bundle/gems/aws_lambda_ric-2.0.0/bin/aws_lambda_ric:10:in `<top (required)>'",
    "/usr/local/bundle/bin/aws_lambda_ric:25:in `load'",
    "/usr/local/bundle/bin/aws_lambda_ric:25:in `<main>'"
  ]
}
END RequestId: 27b5c17f-7b42-4dcf-85d2-3df84810bea1
REPORT RequestId: 27b5c17f-7b42-4dcf-85d2-3df84810bea1	Init Duration: 0.11 ms	Duration: 1068.71 ms	Billed Duration: 1069 ms	Memory Size: 3072 MB	Max Memory Used: 3072 MB	
{"errorMessage":"The security token included in the request is invalid.","errorType":"Function<Aws::SecretsManager::Errors::UnrecognizedClientException>","stackTrace":["/usr/local/bundle/gems/aws-sdk-core-3.131.1/lib/seahorse/client/plugins/raise_response_errors.rb:17:in `call'","/usr/local/bundle/gems/aws-sdk-core-3.131.1/lib/aws-sdk-core/plugins/checksum_algorithm.rb:111:in `call'","/usr/local/bundle/gems/aws-sdk-core-3.131.1/lib/aws-sdk-core/plugins/jsonvalue_converter.rb:22:in `call'","/usr/local/bundle/gems/aws-sdk-core-3.131.1/lib/aws-sdk-core/plugins/idempotency_token.rb:19:in `call'","/usr/local/bundle/gems/aws-sdk-core-3.131.1/lib/aws-sdk-core/plugins/param_converter.rb:26:in `call'","/usr/local/bundle/gems/aws-sdk-core-3.131.1/lib/seahorse/client/plugins/request_callback.rb:71:in `call'","/usr/local/bundle/gems/aws-sdk-core-3.131.1/lib/aws-sdk-core/plugins/response_paging.rb:12:in `call'","/usr/local/bundle/gems/aws-sdk-core-3.131.1/lib/seahorse/client/plugins/response_target.rb:24:in `call'","/usr/local/bundle/gems/aws-sdk-core-3.131.1/lib/seahorse/client/request.rb:72:in `send_request'","/usr/local/bundle/gems/aws-sdk-secretsmanager-1.62.0/lib/aws-sdk-secretsmanager/client.rb:1323:in `get_secret_value'","/function/secrets_manager.rb:39:in `read_secret_json'","/function/secrets_manager.rb:15:in `initialize'","/function/lambda.rb:15:in `new'","/function/lambda.rb:15:in `initialize'","/function/lambda.rb:46:in `new'","/function/lambda.rb:46:in `lambda_handler'","/usr/local/bundle/gems/aws_lambda_ric-2.0.0/lib/aws_lambda_ric/lambda_handler.rb:28:in `call_handler'","/usr/local/bundle/gems/aws_lambda_ric-2.0.0/lib/aws_lambda_ric.rb:81:in `run_user_code'","/usr/local/bundle/gems/aws_lambda_ric-2.0.0/lib/aws_lambda_ric.rb:59:in `start_runtime_loop'","/usr/local/bundle/gems/aws_lambda_ric-2.0.0/lib/aws_lambda_ric.rb:42:in `run'","/usr/local/bundle/gems/aws_lambda_ric-2.0.0/lib/aws_lambda_ric/bootstrap.rb:35:in `bootstrap_handler'","/usr/local/bundle/gems/aws_lambda_ric-2.0.0/lib/aws_lambda_ric/bootstrap.rb:8:in `start'","/usr/local/bundle/gems/aws_lambda_ric-2.0.0/bin/aws_lambda_ric:10:in `<top (required)>'","/usr/local/bundle/bin/aws_lambda_ric:25:in `load'","/usr/local/bundle/bin/aws_lambda_ric:25:in `<main>'"]}2022-06-14 22:42:45,880 | Cleaning all decompressed code dirs
2022-06-14 22:42:45,881 | Sending Telemetry: {'metrics': [{'commandRun': {'requestId': '30b28989-9feb-4a2f-870f-a7c09a293b99', 'installationId': '0b0f3b2b-3efb-438e-b497-cf6bf622fdf6', 'sessionId': 'fbf696f2-5466-43d2-bbc7-8505ad005ea3', 'executionEnvironment': 'CLI', 'ci': False, 'pyversion': '3.7.10', 'samcliVersion': '1.50.0', 'awsProfileProvided': True, 'debugFlagProvided': True, 'region': '', 'commandName': 'sam local invoke', 'metricSpecificAttributes': {'projectType': 'CFN'}, 'duration': 1521, 'exitReason': 'success', 'exitCode': 0}}]}
2022-06-14 22:42:46,510 | HTTPSConnectionPool(host='aws-serverless-tools-telemetry.us-west-2.amazonaws.com', port=443): Read timed out. (read timeout=0.1)
2022-06-14 22:42:46,511 | Telemetry endpoint configured to be https://aws-serverless-tools-telemetry.us-west-2.amazonaws.com/metrics
2022-06-14 22:42:46,511 | Sending Telemetry: {'metrics': [{'runtimeMetric': {'requestId': '83b04abb-a537-489d-a6e7-c4ba9d515844', 'installationId': '0b0f3b2b-3efb-438e-b497-cf6bf622fdf6', 'sessionId': 'fbf696f2-5466-43d2-bbc7-8505ad005ea3', 'executionEnvironment': 'CLI', 'ci': False, 'pyversion': '3.7.10', 'samcliVersion': '1.50.0', 'runtimes': [None]}}]}
2022-06-14 22:42:47,130 | HTTPSConnectionPool(host='aws-serverless-tools-telemetry.us-west-2.amazonaws.com', port=443): Read timed out. (read timeout=0.1)

```

### Expected result:
Response from my function

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: Linux
2. `sam --version`: SAM CLI, version 1.50.0
3. AWS region: eu-west-2

**Comment by qingchm** [comment]
Thanks for reporting in this issue! To further understand the issue, if you list the emulation images, which tag/pull date does your image have?

**Comment by darioackermann** [comment]
Experiencing the same issue. Have you been able to resolve it?

**Comment by sriram-mv** [comment]
I can't seem to reproduce this, do you have the latest images for the emulation images? could you update to the latest version of SAM CLI and run the same build process and local invoke without supplying `--skip-pull-image`. This will force SAM CLI to pull down the latest emulation images.

**Comment by darioackermann** [comment]
Thanks for getting back to me.

We eventually resolved the issue (it was on our side).

-  We have used `boto3` (so not exactly AWS SAM CLI - I posted here because this was one of the only issues to this topic) 
-  We inited a `boto3` session, giving an alternative option to supply access identity / secret. (via kwargs)
   - This worked well locally since our devs were using permanent credentials.
   - The Lambda function however, assumes its role and gains temporary credentials (requiring a session token). 
 - Now, when initing a boto3 session with an access id / secret, `botocore` will completely reset the internal credentials. (including the session token it previously acquired by assuming the role) 
 - Since we had no reason to set any session_token locally, `botocore` takes the default parameter value (`None`) and sets it as part of the credentials (this can be seen here: https://github.com/boto/botocore/blob/428d8c088cac4e9f148c1d9a8b6b95180a39adef/botocore/session.py#L936 )
   - That was the reason that it worked locally but not on Lambda (or any environment with permissions assumed by role)
   
**Removing the option to manually set an access identity / secret resolved the case for us.**

I'm writing this since this might be helpful to anyone in the future.
